packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -225,6 +236,57 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
   get isJavaScript() {
     return Boolean(this.entrySourceFilePath?.endsWith('.js') || this.entrySourceFilePath?.endsWith('.ts'))
   }
+
+  get functionExtension(): FunctionExtension | undefined {
+    if (!this.isFunctionExtension) return undefined
+    return this as unknown as FunctionExtension
+  }
+
+  async buildStep(options: ExtensionBuildOptions) {
+    if (this.isThemeExtension) {
+      return buildThemeExtension(this, options)
+    } else if (this.isFunctionExtension) {
+      return buildFunctionExtension(this.functionExtension!, options)
+    } else if (this.features.includes('esbuild')) {
+      return buildUIExtension(this, options)
+    }
+
+    // Workaround for tax_calculations because they remote spec NEEDS a valid js file to be included.
+    if (this.type === 'tax_calculation') {
```

Comment:
Could we not fix the remote spec?

Reply:
they are going to, but will take some time, in the meantime i'll keep this here and remove it later